spec/invalid-tree-spec.js

```diff
@@ -0,0 +1,120 @@
+const LTT = require('../dist/list-to-tree');
+
+describe('Invalid list:', function() {
+
+    let tree = null;
+
+    const key_id = 'id';
+
+    const key_parent = 'parent';
+
+    const key_child = 'child';
+
+    const depthKeyName = '__depth'
+
+    const consoleLogOfSilence = (func) => {
+        const log = console.log;
+        console.log = () => {};
+        func();
+        console.log = log;
+    }
+
+    beforeEach(function() {
+        const list = [
+            {
+                id: 1,
+                parent: 0
+            }, {
+                id: 20,
+                parent: 1
+            }, {
+                id: 30,
+                parent: 1
+            }, {
+                id: 14,
+                parent: 20
+            }, {
+                id: 15,
+                parent: 20
+            }, {
+                id: 6,
+                parent: 14
+            }, {
+                id: 7,
+                parent: 14
+            }, {
+                id: 8,
+                parent: 100 // bad id
+            }, {
+                id: 9,
+                parent: 101 // bad id
+            }, {
+                id: 10,
+                parent: 102 // bad id
```

Comment:
I also added a test if it was an invalid list.
(To prevent the recursive function from going into an infinite loop)